ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetLogsTest.java

```diff
@@ -0,0 +1,296 @@
+
+    when(blockchainQueries.matchingLogs(anyLong(), anyLong(), any(), any()))
+        .thenReturn(new ArrayList<>());
+
+    final JsonRpcResponse response = method.response(request);
+    assertThat(response).isInstanceOf(JsonRpcSuccessResponse.class);
+
+    verify(blockchainQueries).matchingLogs(eq(genesisBlock), eq(latestBlock), any(), any());
+    verify(blockchainQueries, never()).finalizedBlockHeader();
+    verify(blockchainQueries, never()).safeBlockHeader();
+  }
+
+  @Test
+  public void shouldQueryWrappedNumeric() {
+    final long fromBlock = 50L;
+    final long toBlock = 100L;
+    final JsonRpcRequestContext request =
+        buildRequest(String.valueOf(fromBlock), String.valueOf(toBlock));
+
+    when(blockchainQueries.matchingLogs(anyLong(), anyLong(), any(), any()))
+        .thenReturn(new ArrayList<>());
+
+    final JsonRpcResponse response = method.response(request);
+    assertThat(response).isInstanceOf(JsonRpcSuccessResponse.class);
+
+    verify(blockchainQueries).matchingLogs(eq(fromBlock), eq(toBlock), any(), any());
+    verify(blockchainQueries, never()).finalizedBlockHeader();
+    verify(blockchainQueries, never()).headBlockNumber();
+    verify(blockchainQueries, never()).safeBlockHeader();
+  }
+
+  @Test
+  public void shouldQueryWrappedNumericToNumeric() {
+    final long fromBlock = 50L;
+    final long toBlock = 100L;
+    final JsonRpcRequestContext request = buildRequest(String.valueOf(fromBlock), toBlock);
+
+    when(blockchainQueries.matchingLogs(anyLong(), anyLong(), any(), any()))
+        .thenReturn(new ArrayList<>());
+
+    final JsonRpcResponse response = method.response(request);
+    assertThat(response).isInstanceOf(JsonRpcSuccessResponse.class);
+
+    verify(blockchainQueries).matchingLogs(eq(fromBlock), eq(toBlock), any(), any());
+    verify(blockchainQueries, never()).finalizedBlockHeader();
+    verify(blockchainQueries, never()).headBlockNumber();
+    verify(blockchainQueries, never()).safeBlockHeader();
+  }
+
+  @Test
+  public void shouldQueryEarliestToLatest() {
+    final long genesisBlock = 0L;
+    final long latestBlock = 50L;
+    final JsonRpcRequestContext request = buildRequest("earliest", "latest");
+
+    when(blockchainQueries.headBlockNumber()).thenReturn(latestBlock);
+    when(blockchainQueries.matchingLogs(anyLong(), anyLong(), any(), any()))
+        .thenReturn(new ArrayList<>());
+
+    final JsonRpcResponse response = method.response(request);
+    assertThat(response).isInstanceOf(JsonRpcSuccessResponse.class);
+
+    verify(blockchainQueries).matchingLogs(eq(genesisBlock), eq(latestBlock), any(), any());
+    verify(blockchainQueries, never()).finalizedBlockHeader();
+    verify(blockchainQueries, never()).safeBlockHeader();
+  }
+
+  @Test
+  public void shouldQuerySafeToFinalized() {
+    final long finalizedBlockNumber = 50L;
+    final long safeBlockNumber = 25L;
+    final JsonRpcRequestContext request = buildRequest("safe", "Finalized");
```

Comment:
is the capitalization of "Finalized" intended? is it case insensitive?

Reply:
Not intentional, but it gets sanitised in BlockParameter. Good catch. 👍